# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by scripts/Create-CopyPatch.
# 
# T2 SDE: package/*/xemacs/hotfix-glibc.patch
# Copyright (C) 2021 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- xemacs-21.5.34/src/gmalloc.c.vanilla	2021-07-14 17:33:06.143824000 +0200
+++ xemacs-21.5.34/src/gmalloc.c	2021-07-14 17:36:04.803824000 +0200
@@ -69,9 +69,9 @@
 #include <string.h>
 #include <limits.h>
 
-#ifdef	HAVE_UNISTD_H
+//#ifdef	HAVE_UNISTD_H
 #include <unistd.h>
-#endif
+//#endif
 
 #endif	/* _MALLOC_INTERNAL.  */
 
@@ -319,10 +319,6 @@
 #if defined (__GNU_LIBRARY__) || defined (_LIBC)
 #include <stddef.h>
 #include <sys/cdefs.h>
-#if ! (defined (__GLIBC__) && (__GLIBC__ >= 2))
-extern size_t __getpagesize __P ((void));
-#endif
-#else
 #include "getpagesize.h"
 #define	 __getpagesize()	getpagesize()
 #endif
@@ -1207,7 +1203,7 @@
 /* It is best not to declare this and cast its result on foreign operating
    systems with potentially hostile include files.  */
 #if !(defined(linux) && defined(sparc))
-extern __ptr_t __sbrk __P ((int increment));
+//extern __ptr_t __sbrk __P ((int increment));
 #endif
 #endif
 
